admission/hooks/mutate_pod.go

```diff
@@ -0,0 +1,98 @@
+package hooks
+
+import (
+	"context"
+	"crypto/sha1"
+	"encoding/hex"
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// +kubebuilder:webhook:verbs=create,path=/mutate-pod,mutating=true,failurePolicy=fail,groups="",resources=pods,versions=v1,name=mpod.kb.io
+
+type podMutator struct {
+	client  client.Client
+	decoder *admission.Decoder
+}
+
+// NewPodMutator creates a webhook handler for Pod.
+func NewPodMutator(c client.Client, dec *admission.Decoder) http.Handler {
+	return &webhook.Admission{Handler: &podMutator{c, dec}}
+}
+
+func (m *podMutator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	po := &corev1.Pod{}
+	err := m.decoder.Decode(req, po)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	poPatched := po.DeepCopy()
+	for i, co := range po.Spec.Containers {
+		if !m.isMountedTmp(&co) {
+			volumeName := m.generateVolumeName(co.Name, po.Spec.Volumes)
+			m.appendEmptyDir(volumeName, poPatched)
+			poPatched.Spec.Containers[i].VolumeMounts = append(poPatched.Spec.Containers[i].VolumeMounts,
+				corev1.VolumeMount{Name: volumeName, MountPath: "/tmp"})
+		}
+	}
+	for i, co := range po.Spec.InitContainers {
+		if !m.isMountedTmp(&co) {
+			volumeName := m.generateVolumeName(co.Name, po.Spec.Volumes)
+			m.appendEmptyDir(volumeName, poPatched)
+			poPatched.Spec.InitContainers[i].VolumeMounts = append(poPatched.Spec.InitContainers[i].VolumeMounts,
+				corev1.VolumeMount{Name: volumeName, MountPath: "/tmp"})
+		}
+	}
+
+	marshaled, err := json.Marshal(poPatched)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	return admission.PatchResponseFromRaw(req.Object.Raw, marshaled)
+}
+
+func (m *podMutator) isMountedTmp(co *corev1.Container) bool {
+	for _, mount := range co.VolumeMounts {
+		if mount.MountPath == "/tmp" || strings.HasPrefix(mount.MountPath, "/tmp/") {
+			return true
+		}
+	}
+	return false
+}
+
+func (m *podMutator) hashString(name string) string {
+	return hex.EncodeToString(sha1.New().Sum([]byte(name)))
```

Comment:
```suggestion
	sum := sha1.Sum([]byte(name))
	return hex.EncodeToString(sum[:])
```

Reply:
I understood.
My fix was wrong. The hash value could not be calculated.